c_crypto.h

```diff
@@ -3,59 +3,44 @@
 
 #include <openssl/aes.h>
 #include <openssl/bio.h>
+#include <openssl/crypto.h>
 #include <openssl/err.h>
 #include <openssl/evp.h>
 #include <openssl/hmac.h>
+#include <openssl/md5.h>
 #include <openssl/pem.h>
 #include <openssl/rand.h>
 #include <openssl/rsa.h>
 #include <openssl/sha.h>
 #include <openssl/ssl.h>
 #include <openssl/x509.h>
 
-#include "c_common.h"
-
-#define AES_CBC_128_KEY_SIZE 128
-#define AES_CBC_128_IV_SIZE 16
-#define DIST_KEY_EXPIRATION_TIME_SIZE 6
-#define KEY_EXPIRATION_TIME_SIZE 6
-#define SESSION_KEY_ID_SIZE 8
+#define AES_128_KEY_SIZE_IN_BYTES 16
+#define AES_128_IV_SIZE 16
+#define AES_128_CBC_IV_SIZE 16
+#define AES_128_CTR_IV_SIZE 16
+#define AES_128_GCM_IV_SIZE 12
+#define AES_GCM_TAG_SIZE 12
 #define ABS_VALIDITY_SIZE 6
 #define REL_VALIDITY_SIZE 6
-#define MAC_KEY_SIZE 32  // FIXME: To be replaced by config.
 #define MAX_MAC_KEY_SIZE 32
+#define MAC_KEY_SHA256_SIZE 32
 #define CIPHER_KEY_SIZE 16  // FIXME: To be replaced by config.
```

Comment:
Please double-check this FIXME.

Reply:
Removed fixme. 99f3ef76c852042205da4cda244c299304dc312a